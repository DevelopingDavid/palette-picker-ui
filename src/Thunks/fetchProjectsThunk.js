import { addProjects, isLoading, hasError } from '../Actions'

export const fetchProjectsThunk = () => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch('http://localhost:3001/api/v1/projects')
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      const projects = await response.json()
      dispatch(isLoading(false))
      dispatch(addProjects(projects))
      return projects
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}

export default fetchProjectsThunk