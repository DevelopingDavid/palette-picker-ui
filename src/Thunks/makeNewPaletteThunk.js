import { isLoading, hasError } from '../Actions'

export const makeNewPaletteThunk = (palette) => {
  return async (dispatch) => {
    const options = {
      method: 'POST',
      body: JSON.stringify(palette),
      headers: {
        "Content-Type": "application/json"
      }
    }
    try {
      dispatch(isLoading(true))
      const response = await fetch(`http://localhost:3001/api/v1/projects/${palette.project_id}/palettes`, options)
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      dispatch(isLoading(false))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}

export default makeNewPaletteThunk;