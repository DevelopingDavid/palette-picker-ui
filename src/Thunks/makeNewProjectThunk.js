import { makeNewProject, isLoading, hasError } from '../Actions'

export const makeNewProjectThunk = (project) => {
  return async (dispatch) => {
    const options = {
      method: "POST",
      body: JSON.stringify(project),
      headers: {
        "Content-Type": "application/json"
      }
    }
    try {
      dispatch(isLoading(true))
      const response = await fetch(`${REACT_APP_BACKEND_URL}/api/v1/projects`, options)
      if (!response.ok) {
        throw new Error(response.statusText)
      }
      const projectId = await response.json()
      dispatch(isLoading(false))
      dispatch(makeNewProject(projectId))
      return projectId
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}

export default makeNewProjectThunk