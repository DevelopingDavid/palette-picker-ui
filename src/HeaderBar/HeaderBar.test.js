import React from 'react'
import { HeaderBar, mapStateToProps, mapDispatchToProps } from './HeaderBar'
import { fetchProjectsThunk } from '../Thunks/fetchProjectsThunk'
jest.mock('../Thunks/fetchProjectsThunk')

describe('HeaderBar', () => {
  describe('mapStateToProps', () => {
    it('should mapStateToProps', () => {
      const mockState = {}
      const upDatedState = {
        projects: []
      }
      const firstState = mapStateToProps(mockState)
      expect(firstState).toEqual({})
      const newState = mapStateToProps(upDatedState)
      expect(upDatedState).toEqual(newState)
    })
  })

  describe.skip('mapDispatchToProps', () => {
    it('should dispatch fetchProjectsThunk', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = fetchProjectsThunk()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchProjects()
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})