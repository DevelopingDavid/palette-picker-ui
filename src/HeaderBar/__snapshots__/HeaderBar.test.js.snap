// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar Should match snapshot 1`] = `
<div
  className=""
>
  <WithStyles(CssBaseline) />
  <WithStyles(AppBar)
    className=""
    position="fixed"
  >
    <WithStyles(Toolbar)
      disableGutters={true}
    >
      <WithStyles(IconButton)
        aria-label="Open drawer"
        className=""
        color="inherit"
        onClick={[Function]}
      >
        <pure(MenuIcon) />
      </WithStyles(IconButton)>
      <WithStyles(Typography)
        color="inherit"
        noWrap={true}
        variant="h6"
      >
        Palette Picker
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(Drawer)
    anchor="left"
    classes={
      Object {
        "paper": undefined,
      }
    }
    open={false}
    variant="persistent"
  >
    <div>
      <WithStyles(IconButton)
        onClick={[Function]}
      >
        <pure(ChevronRightIcon) />
      </WithStyles(IconButton)>
    </div>
    <WithStyles(Divider) />
    <WithStyles(List)>
      <ConnectFunction
        key="50"
        project={
          Object {
            "id": 50,
            "project_name": "My Project",
          }
        }
      />
    </WithStyles(List)>
  </WithStyles(Drawer)>
  <main
    className=""
  >
    <div />
  </main>
</div>
`;
